LX.StaffScheduler.DAL/Repositories/WorkShiftRepository.cs

```diff
@@ -40,5 +40,33 @@ public async Task RemoveAsync(int id)
             _context.WorkShifts.Remove(workShift);
             await _context.SaveChangesAsync();
         }
+
+        public async Task<IEnumerable<DateOnly>> GetMondaysWorkShiftsAsync(int cafeId)
+        {
+            var weekStartDates = new List<DateOnly>();
+
+            var workShifts = await _context.WorkShifts
+                .Where(ws => ws.CafeId == cafeId && ws.ShiftDate.DayOfWeek == DayOfWeek.Monday)
+                .OrderBy(ws => ws.ShiftDate)
+                .ToListAsync();
+
+            var oldestMondayDate = workShifts.Select(ws => ws.ShiftDate).FirstOrDefault();
+
+            if (oldestMondayDate == default)
```

Comment:
not clear with this, could you explain?

Reply:
right in this place we select the oldest monday date. if it equals default value for DateOnly type (0001-01-01, if i correct), then we understand that there is no mondays in workshifts